### PR TITLE
fix: Removed GSON jar dependency and added plugin dependency

### PR DESCRIPTION
**Pull request by Jagrutiti**
### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

Added the plugin dependency GSON library as mentioned in the issue. 

Closes 
[#JENKINS-72494](https://issues.jenkins.io/projects/JENKINS/issues/JENKINS-72494)

### CODE REVIEW

**Comment by uhafner** [comment]
Thanks for your PR. I am not planning to change the Jenkins baseline right now. I will revive this PR when the baseline has been moved…

File: plugin/pom.xml [hunk 1]
```diff
@@ -51,6 +51,7 @@
     <argLine>-Djava.awt.headless=true -Xmx1024m -Djenkins.test.timeout=1000 --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.io=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.util.concurrent=ALL-UNNAMED</argLine>
 
     <prism-api.version>1.29.0-8</prism-api.version>
+    <jenkins.version>2.414.1</jenkins.version>
```
Comment (jonesbusy) [inline plugin/pom.xml:54]: gson-api need minimum 2.401.3. Not 2.414.1

File: plugin/pom.xml [hunk 2]
```diff
@@ -181,9 +182,9 @@
       <artifactId>apache-httpcomponents-client-4-api</artifactId>
     </dependency>
     <dependency>
-      <groupId>com.google.code.gson</groupId>
-      <artifactId>gson</artifactId>
-      <version>2.10.1</version>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>gson-api</artifactId>
+      <version>2.10.1-15.v0d99f670e0a_7</version>
```
Comment (jonesbusy) [inline plugin/pom.xml:187]: Don't include version if part of bom

Comment (Jagrutiti) [reply plugin/pom.xml:187]: If I remove the version I get the following error:

```
[ERROR]   The project io.jenkins.plugins:warnings-ng:10.7.0-SNAPSHOT (C:\Users\tiwar\open-source\warnings-ng-plugin\plugin\pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for io.jenkins.plugins:gson-api:jar is missing. @ io.jenkins.plugins:warnings-ng:${revision}${changelist}, 
```

File: plugin/pom.xml (hunk 1 above)
Comment (Jagrutiti) [reply plugin/pom.xml:54]: I had done tried both the comments.

In this case, I get the error:

```
[ERROR] Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.50:validate-hpi (default-validate-hpi) on project warnings-ng: Dependency io.jenkins:configuration-as-code:jar:1714.v09593e830cfa requires Jenkins 2.414.1 or higher.
```

File: plugin/pom.xml (hunk 2 above)
Comment (jonesbusy) [reply plugin/pom.xml:187]: So use a bom version that include gson-api. Is part of bom since something like 1 month

**Pull request closed without merging**